client.go

```diff
@@ -636,23 +630,30 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 		}
 
 		wait := c.Backoff(c.RetryWaitMin, c.RetryWaitMax, i, resp)
-		desc := fmt.Sprintf("%s %s", req.Method, req.URL)
-		if code > 0 {
-			desc = fmt.Sprintf("%s (status: %d)", desc, code)
-		}
 		if logger != nil {
+			var code int // HTTP response code
+			if resp != nil {
+				code = resp.StatusCode
+			}
+			var desc string
+			if code > 0 {
+				desc = fmt.Sprintf("%s (status: %d)", desc, code)
+			} else {
+				desc = fmt.Sprintf("%s %s", req.Method, req.URL)
+			}
 			switch v := logger.(type) {
 			case LeveledLogger:
 				v.Debug("retrying request", "request", desc, "timeout", wait, "remaining", remain)
 			case Logger:
 				v.Printf("[DEBUG] %s: retrying in %s (%d left)", desc, wait, remain)
 			}
 		}
+		timer := time.NewTimer(wait)
 		select {
 		case <-req.Context().Done():
-			c.HTTPClient.CloseIdleConnections()
```

Comment:
Same comment as above - this was a very intentional design choice so I'm wary of changing the keepalive behavior in this PR unless there is a very good reason.